.github/workflows/main.yml

```diff
@@ -0,0 +1,34 @@
+name: coursist unit tests and linting
+on:
+  push:
+    branches: 
+    - master
+    - develop
+  pull_request:
+    branches: 
+    - master
+    - develop
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install python dependencies
+      run: |
+        pip install -r requirements-dev.txt
+        pip install -r requirements.txt
+    - name: Black Code Formatter
+      uses: lgeiger/black-action@v1.0.1
+      with:
+        args: ". --check -l 120"
```

Comment:
There is no need to run black on the CI. What we need is to make sure black doesn't change any file (thus making sure the code on the branch is already black-formatted).